accord-core/src/main/java/accord/utils/async/AsyncChains.java

```diff
@@ -414,7 +414,7 @@ private AccumulatingReducer(Reduce<V, V> reducer)
             this.reducer = reducer;
         }
 
-        public void add(AsyncChain<V> add)
+        public synchronized void add(AsyncChain<V> add)
```

Comment:
Could you elaborate how this `synchnornized` is enough? `size` and `start` are still unsynchronized; 
I think it might be good to describe visibility guarantee, ie from what I understand we rely on volatile + snapshot of `state` with a single writer.

Reply:
I didn't end up using it here so I could just revert, but to explain: the only thing we worry about here is collecting items to reduce in the list, which can safely be done asynchronously if this method is synchronized. We must already assume that all of the items we want to collect are already collected when we start the chain, and that this is done by a single owner, otherwise the whole thing is sort of broken at a more fundamental level.